### ISSUE TITLE
Hatch install failing in Ubuntu Jammy dev container created via VSCode/Docker on macOS

### ISSUE THREAD

**Issue by amosaikman** [devcontainers-contrib/features#279]
Steps to reproduce:

> Dev Containers: Create Dev Container -> Ubunutu (default/Jammy) -> select Python and Hatch (only in this example)

Full log below. 

```
[58 ms] Dev Containers 0.266.1 in VS Code 1.74.3 (97dec172d3256f8ca4bfb2143f3f76b503ca0534).
[57 ms] Start: Resolving Remote
[66 ms] Start: Check Docker is running
[67 ms] Start: Run: docker version --format {{.Server.APIVersion}}
[147 ms] Server API version: 1.41
[148 ms] Start: Run: docker build -f /var/folders/09/7s45bd2s62l9t17yh52vh2h00000gn/T/vsch/bootstrap-image/0.266.1/bootstrap.Dockerfile -t vsc-volume-bootstrap /var/folders/09/7s45bd2s62l9t17yh52vh2h00000gn/T/vsch/bootstrap-image/0.266.1
[+] Building 0.7s (9/9) FINISHED                                                
 => [internal] load build definition from bootstrap.Dockerfile             0.0s
 => => transferring dockerfile: 84B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for mcr.microsoft.com/devcontainers/base:0-a  0.5s
 => [1/4] FROM mcr.microsoft.com/devcontainers/base:0-alpine-3.16@sha256:  0.0s
 => [internal] load build context                                          0.1s
 => => transferring context: 7.46kB                                        0.0s
 => CACHED [2/4] RUN apk add --no-cache  nodejs  python3  npm  make  g++   0.0s
 => CACHED [3/4] RUN cd && npm i node-pty                                  0.0s
 => CACHED [4/4] COPY .vscode-remote-containers /root/.vscode-remote-cont  0.0s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:1e2e90c4a334d69de3d574b0203bd8b27d8976eff959a  0.0s
 => => naming to docker.io/library/vsc-volume-bootstrap                    0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
[1562 ms] Start: Run: docker run -d --mount type=volume,src=ubuntu,dst=/workspaces -v /var/run/docker.sock:/var/run/docker.sock vsc-volume-bootstrap sleep infinity
[2080 ms] Start: Run in container: /bin/sh
[2482 ms] Start: Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js read-configuration --workspace-folder /workspaces/ubuntu --id-label vsc.devcontainer.volume.name=ubuntu --id-label vsc.devcontainer.volume.folder=ubuntu --log-level debug --log-format json --mount-workspace-git-root true
[3067 ms] (node:25) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[3067 ms] (Use `node --trace-deprecation ...` to show where the warning was created)
[3074 ms] @devcontainers/cli 0.25.2. Node.js v16.17.1. linux 5.15.49-linuxkit x64.
[3074 ms] Start: Run: git rev-parse --show-cdup
[3085 ms] Start: Run: docker ps -q -a --filter label=vsc.devcontainer.volume.name=ubuntu --filter label=vsc.devcontainer.volume.folder=ubuntu
[3142 ms] Start: Run in container: cat <<'EOF-/tmp/devcontainer-d67537e4-a038-434c-8f53-9f3fc4ab04e9.json' >/tmp/devcontainer-d67537e4-a038-434c-8f53-9f3fc4ab04e9.json_1674449593724
[3147 ms] 
[3147 ms] 
[3148 ms] Start: Run in container: docker volume ls -q
[3643 ms] Start: Run in container: docker ps -q -a --filter label=vsc.devcontainer.volume.name=ubuntu --filter label=vsc.devcontainer.volume.folder=ubuntu
[3823 ms] Start: Run in container: docker ps -q -a --filter label=vsc.devcontainer.volume.name=ubuntu --filter label=vsc.devcontainer.volume.folder=ubuntu
[4327 ms] Start: Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js up --workspace-folder /workspaces/ubuntu --workspace-mount-consistency cached --id-label vsc.devcontainer.volume.name=ubuntu --id-label vsc.devcontainer.volume.folder=ubuntu --log-level debug --log-format json --config /workspaces/ubuntu/.devcontainer/devcontainer.json --override-config /tmp/devcontainer-d67537e4-a038-434c-8f53-9f3fc4ab04e9.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=ubuntu,target=/workspaces,external=true --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default off --mount-workspace-git-root true --terminal-columns 352 --terminal-rows 19
[4840 ms] (node:114) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
[4840 ms] (Use `node --trace-deprecation ...` to show where the warning was created)
[4843 ms] @devcontainers/cli 0.25.2. Node.js v16.17.1. linux 5.15.49-linuxkit x64.
[4843 ms] Start: Run: docker buildx version
[4939 ms] github.com/docker/buildx v0.8.2 6224def4dd2c3d347eee19db595348c50d7cb491
[4940 ms] 
[4940 ms] Start: Resolving Remote
[4945 ms] Start: Run: git rev-parse --show-cdup
[4953 ms] Start: Run: docker ps -q -a --filter label=vsc.devcontainer.volume.name=ubuntu --filter label=vsc.devcontainer.volume.folder=ubuntu
[4993 ms] Start: Run: docker inspect --type image mcr.microsoft.com/devcontainers/base:jammy
[6471 ms] local container features stored at: /root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/node_modules/vscode-dev-containers/container-features
[6474 ms] Start: Run: tar --no-same-owner -x -f -
[6593 ms] * Processing feature: ghcr.io/devcontainers/features/python:1
[7873 ms] * Processing feature: ghcr.io/devcontainers-contrib/features/hatch:2
[9007 ms] * Fetching feature: python_1_oci
[10614 ms] * Fetching feature: hatch_2_oci
[12256 ms] Start: Run: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.25.2-1674449597050 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/base:jammy --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-ubuntu-481f4cbe471f584ee514d6bb2fc80ed7-features -f /tmp/devcontainercli-root/container-features/0.25.2-1674449597050/Dockerfile.extended /tmp/devcontainercli-root/empty-folder
[+] Building 7.7s (15/16)                                                                                                                                                                                                                                                                                                                                       
 => [internal] load build definition from Dockerfile.extended                                                                                                                                                                                                                                                                                              0.0s
 => => transferring dockerfile: 2.31kB                                                                                                                                                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                          0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                                                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1.4                                                                                                                                                                                                                                                                                               3.1s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc                                                                                                                                                                                                                          0.0s
 => [internal] load build definition from Dockerfile.extended                                                                                                                                                                                                                                                                                              0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                                                                          0.0s
 => [context dev_containers_feature_content_source] load .dockerignore                                                                                                                                                                                                                                                                                     0.0s
 => => transferring dev_containers_feature_content_source: 2B                                                                                                                                                                                                                                                                                              0.0s
 => [internal] load metadata for mcr.microsoft.com/devcontainers/base:jammy                                                                                                                                                                                                                                                                                0.2s
 => [context dev_containers_feature_content_source] load from client                                                                                                                                                                                                                                                                                       0.1s
 => => transferring dev_containers_feature_content_source: 311.10kB                                                                                                                                                                                                                                                                                        0.0s
 => [dev_containers_feature_content_normalize 1/3] FROM mcr.microsoft.com/devcontainers/base:jammy@sha256:e86aca6eadad566af771a078919c97e32c80fae65369e1cee9af2e2261fbed85                                                                                                                                                                                 0.0s
 => CACHED [dev_containers_feature_content_normalize 2/3] COPY --from=dev_containers_feature_content_source . /tmp/build-features/                                                                                                                                                                                                                         0.0s
 => CACHED [dev_containers_feature_content_normalize 3/3] RUN chmod -R 0700 /tmp/build-features                                                                                                                                                                                                                                                            0.0s
 => CACHED [dev_containers_target_stage 2/5] COPY --from=dev_containers_feature_content_normalize /tmp/build-features /tmp/build-features                                                                                                                                                                                                                  0.0s
 => CACHED [dev_containers_target_stage 3/5] RUN echo "_CONTAINER_USER_HOME=$(getent passwd root | cut -d: -f6)" >> /tmp/build-features/devcontainer-features.builtin.env && echo "_REMOTE_USER_HOME=$(getent passwd vscode | cut -d: -f6)" >> /tmp/build-features/devcontainer-features.builtin.env                                                       0.0s
 => ERROR [dev_containers_target_stage 4/5] RUN cd /tmp/build-features/hatch_2 && chmod +x ./devcontainer-features-install.sh && ./devcontainer-features-install.sh                                                                                                                                                                                        3.9s
------
 > [dev_containers_target_stage 4/5] RUN cd /tmp/build-features/hatch_2 && chmod +x ./devcontainer-features-install.sh && ./devcontainer-features-install.sh:
#0 0.271 ===========================================================================
#0 0.271 Feature       : Hatch (via pipx)
#0 0.271 Description   : Hatch is a modern, extensible Python project manager.
#0 0.271 Id            : ghcr.io/devcontainers-contrib/features/hatch
#0 0.271 Version       : 2.0.4
#0 0.271 Documentation : http://github.com/devcontainers-contrib/features/tree/main/src/hatch
#0 0.271 Options       :
#0 0.271     VERSION="latest"
#0 0.271 ===========================================================================
#0 3.831 featmake: OK
#0 3.885 ID_LIKE=debian
#0 3.887 /usr/local/bin/featmake: line 27: apk: command not found
#0 3.892 ERROR: Feature "Hatch (via pipx)" (ghcr.io/devcontainers-contrib/features/hatch) failed to install! Look at the documentation at http://github.com/devcontainers-contrib/features/tree/main/src/hatch for help troubleshooting this error.
------
error: failed to solve: executor failed running [/bin/sh -c cd /tmp/build-features/hatch_2 && chmod +x ./devcontainer-features-install.sh && ./devcontainer-features-install.sh]: exit code: 127
[20424 ms] Error: Command failed: docker buildx build --load --build-context dev_containers_feature_content_source=/tmp/devcontainercli-root/container-features/0.25.2-1674449597050 --build-arg _DEV_CONTAINERS_BASE_IMAGE=mcr.microsoft.com/devcontainers/base:jammy --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp --target dev_containers_target_stage -t vsc-ubuntu-481f4cbe471f584ee514d6bb2fc80ed7-features -f /tmp/devcontainercli-root/container-features/0.25.2-1674449597050/Dockerfile.extended /tmp/devcontainercli-root/empty-folder
[20424 ms]     at Aoe (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:1893:1355)
[20424 ms]     at uT (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:1893:1291)
[20425 ms]     at processTicksAndRejections (node:internal/process/task_queues:96:5)
[20425 ms]     at async Poe (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:1899:2128)
[20425 ms]     at async Zf (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:1899:3278)
[20425 ms]     at async aue (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:2020:15276)
[20425 ms]     at async oue (/root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js:2020:15030)
[20462 ms] Exit code 1
[20462 ms] Start: Run: docker rm -f 3f98f8c3f44ba52b914c552c8eab89faf17f30608f05ffbbe6048f0e699e2854
[20468 ms] Command failed: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.266.1/dist/spec-node/devContainersSpecCLI.js up --workspace-folder /workspaces/ubuntu --workspace-mount-consistency cached --id-label vsc.devcontainer.volume.name=ubuntu --id-label vsc.devcontainer.volume.folder=ubuntu --log-level debug --log-format json --config /workspaces/ubuntu/.devcontainer/devcontainer.json --override-config /tmp/devcontainer-d67537e4-a038-434c-8f53-9f3fc4ab04e9.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=ubuntu,target=/workspaces,external=true --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default off --mount-workspace-git-root true --terminal-columns 352 --terminal-rows 19
[20468 ms] Exit code 1
```

**Comment by arrrgi** [comment]
@amosaikman can you try again now. I think this should be fixed by #282 which was merged yesterday

**Comment by amosaikman** [comment]
Thanks @arrrgi, seems to be fixed now.
